src/tasks/AotCompilerTask/MonoAOTCompiler.cs

```diff
@@ -305,12 +305,63 @@ public class MonoAOTCompiler : Microsoft.Build.Utilities.Task
     private FileCache? _cache;
     private int _numCompiled;
     private int _totalNumAssemblies;
+    private bool _collectTrimmingEligibleMethodsValueFromPreviousBuild;
 
     private readonly Dictionary<string, string> _symbolNameFixups = new();
     private static readonly char[] s_semicolon = new char[]{ ';' };
 
     private bool ProcessAndValidateArguments()
     {
+        string monoAotPropertyValuesFilePath = Path.Combine(IntermediateOutputPath, "monoAotPropertyValues.txt");
+
+        if (File.Exists(monoAotPropertyValuesFilePath))
+        {
+            string? text = File.ReadAllText(monoAotPropertyValuesFilePath);
+            if (!string.IsNullOrEmpty(text))
+            {
+                Dictionary<string, bool>? properties = JsonSerializer.Deserialize<Dictionary<string, bool>>(text);
+                if ((properties != null) && properties.ContainsKey("CollectTrimmingEligibleMethods"))
+                    if (properties.TryGetValue("CollectTrimmingEligibleMethods", out bool value))
+                        _collectTrimmingEligibleMethodsValueFromPreviousBuild = value;
+            }
+            if (_collectTrimmingEligibleMethodsValueFromPreviousBuild != CollectTrimmingEligibleMethods)
+                File.Delete(monoAotPropertyValuesFilePath);
+        }
+
+        if (!File.Exists(monoAotPropertyValuesFilePath) || ((_collectTrimmingEligibleMethodsValueFromPreviousBuild != CollectTrimmingEligibleMethods)))
```

Comment:
- separate method that is not specific to `CollectTrimmingEligibleMethods`

Reply:
done